src/core/defs.jl

```diff
@@ -578,13 +578,44 @@ unit(obj::ParameterDefReference) = parameter(obj).unit
     variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::ComponentPath, var_name::Symbol)
 
 Return the names of the dimensions of variable `var_name` exposed in the composite 
-component definition indicated by`obj` along the component path `comp_path`.
+component definition indicated by`obj` along the component path `comp_path`. The 
+`comp_path`is of type `Mimi.ComponentPath` with the single arugment being a named
+tuple of symbols describing the relative (to a composite) or absolute (relative to ModelDef) 
+path through composite nodes to specific composite or leaf node.
```

Comment:
saying "a named tuple" confused me at first, cause it's actually just a Tuple right?

Reply:
Hm I could reword this slightly, but I had copied it from @rjplevin's response on the forum. For the `ComponentPath` type, the single **field**, not **argument**, is of type `NTuple` (named tuple) which is not the same as `Tuple`.
```
# ComponentPath identifies the path through multiple composites to a leaf comp.
struct ComponentPath <: MimiStruct
    names::NTuple{N, Symbol} where N
end
```
That said, the constructor allows you to input a `Tuple` as the argument, which it will convert to `NTuple`, as shown by something like:
```
function variable_dimensions(obj::AbstractCompositeComponentDef, comp::Symbol, var_name::Symbol)
    return variable_dimensions(obj, Mimi.ComponentPath((comp,)), var_name)
end
```

Reply:
The doctoring could instead read something like:
"Return the names of the dimensions of variable `var_name` exposed in the composite component definition indicated by`obj` along the component path `comp_path`. The  `comp_path`is of type `Mimi.ComponentPath` with a single field, a Named Tuple of symbols describing the relative (to a composite) or absolute (relative to ModelDef) path through composite nodes to specific composite or leaf node. The constructor for a `ComponentPath` can, however, take a regular `Tuple`. 

@rjplevin what do you think the right wording is here?

Reply:
I think @rjplevin's comment below clears this up for me so I can fix it, as you said @ckingdon95 it's not a `NamedTuple.